src/test/java/org/jenkinsci/plugins/authorizeproject/strategy/SystemAuthorizationStrategyTest.java

```diff
@@ -0,0 +1,447 @@
+                new NullInputStream(0),
+                stdout,
+                stderr
+            );
+            assertEquals(stderr.toString(), 0, ret);
+            configXml = stdout.toString();
+        }
+        
+        // POST config.xml of srcProject (userid is set to admin) to a new project.
+        // This should fail.
+        FreeStyleProject destProject = j.createFreeStyleProject();
+        destProject.save();
+        String projectName = destProject.getFullName();
+        
+        {
+            CLI cli = new CLI(j.getURL());
+            ByteArrayOutputStream stdout = new ByteArrayOutputStream();
+            ByteArrayOutputStream stderr = new ByteArrayOutputStream();
+            int ret = cli.execute(Arrays.asList(
+                    "update-job",
+                    destProject.getFullName(),
+                    "--username",
+                    "test1",
+                    "--password",
+                    "test1"
+                ),
+                new ByteArrayInputStream(configXml.getBytes()),
+                stdout,
+                stderr
+            );
+            assertNotEquals(0, ret);
+        }
+        
+        {
+            FreeStyleProject p = j.jenkins.getItemByFullName(projectName, FreeStyleProject.class);
+            assertNotNull(p);
+            AuthorizeProjectProperty prop = p.getProperty(AuthorizeProjectProperty.class);
+            assertThat(prop, anyOf(nullValue(), hasProperty("strategy", nullValue())));
+        }
+        
+        j.jenkins.reload();
+        
+        {
+            FreeStyleProject p = j.jenkins.getItemByFullName(projectName, FreeStyleProject.class);
+            assertNotNull(p);
+            AuthorizeProjectProperty prop = p.getProperty(AuthorizeProjectProperty.class);
+            assertThat(prop, anyOf(nullValue(), hasProperty("strategy", nullValue())));
+        }
+    }
+    
+    @Test
+    public void testConfigurationAuthentication() throws Exception {
+        prepareSecurity();
+        
+        FreeStyleProject p = j.createFreeStyleProject();
+        
+        WebClient wc = j.createWebClient();
+        wc.login("test1");
+
+        SystemAuthorizationStrategy.DescriptorImpl descriptor =
+                j.getInstance().getDescriptorByType(SystemAuthorizationStrategy.DescriptorImpl.class);
+        p.addProperty(new AuthorizeProjectProperty(new SystemAuthorizationStrategy()));
+
+        // Reconfiguration is allowed if reconfiguration is permitted.
+        descriptor.setPermitReconfiguration(true);
+        j.submit(wc.getPage(p, "configure").getFormByName("config"));
+
+        // Reconfiguration is not allowed if reconfiguration is permitted.
+        descriptor.setPermitReconfiguration(false);
+        try {
+            j.submit(wc.getPage(p, "configure").getFormByName("config"));
+            fail();
+        } catch (FailingHttpStatusCodeException e) {
+            assertEquals(400, e.getStatusCode());
+        }
+    }
+    
+
+}
```

Comment:
Tests look great (no clue what they're doing). NIT: Looks like they could be dried out a bit though - fair amount of duplication around calls to the CLI.